spec/unit/actions/synchronize_user_groups_spec.rb

```diff
@@ -0,0 +1,61 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata/lcm/lcm2'
+require 'gooddata/lcm/actions/synchronize_user_groups'
+
+describe GoodData::LCM2::SynchronizeUserGroups do
+  let(:gdc_gd_client) { double(:gdc_gd_client) }
+  let(:development_client) { double(:development_client) }
+  let(:logger) { double(:logger) }
+
+  let(:from_project_user_group) { double(:from_project_user_group) }
+
+  let(:from_project) { double(:from_project) }
+  let(:to_project) { double(:to_project) }
+
+  let(:params) do
+    params = {
+      gdc_gd_client: gdc_gd_client,
+      development_client: development_client,
+      synchronize: [{
+        segment_id: 'some_segment_ids',
+        from: 'from_project_id',
+        to: [{ pid: 'to_project_id' }]
+      }],
+      gdc_logger: logger
+    }
+    GoodData::LCM2.convert_to_smart_hash(params)
+  end
+
+  before do
+    allow(logger).to receive(:info)
+    allow(from_project_user_group).to receive(:name) { 'user_group_name' }
+    allow(from_project_user_group).to receive(:description) { 'user_group_description' }
+    allow(from_project_user_group).to receive(:project=)
+    allow(from_project_user_group).to receive(:description=)
+    allow(from_project_user_group).to receive(:save)
+
+    allow(development_client).to receive(:projects) { from_project }
+    allow(from_project).to receive(:user_groups) { [from_project_user_group] }
+    allow(from_project).to receive(:title)
+    allow(from_project).to receive(:pid) { 'from_project_id' }
+    allow(from_project).to receive(:dashboards) { [] }
+
+    allow(gdc_gd_client).to receive(:projects) { to_project }
+    allow(gdc_gd_client).to receive(:create) { from_project_user_group }
+    allow(to_project).to receive(:user_groups) { [] }
+    allow(to_project).to receive(:title)
+    allow(to_project).to receive(:uri)
+    allow(to_project).to receive(:pid) { 'to_project_id' }
+    allow(to_project).to receive(:client) { gdc_gd_client }
+  end
+
+  it 'should transfer user groups' do
+    results = subject.class.call(params)
+    expect(results).to eq [{ from: 'from_project_id', to: 'to_project_id', user_group: 'user_group_name', status: 'created' }]
+  end
```

Comment:
Please add a test for transferring the groups: `expect(GoodData::Project).to receive(:transfer_user_groups).with(...)`.